### PR TITLE
snickers: NICPS-32: add function to set appointment color

### PR DESCRIPTION
**Pull request by k-kato**
See the following pull requests: 
https://github.com/Zimbra/zm-mailbox/pull/564
https://github.com/Zimbra/zm-web-client/pull/250

### CODE REVIEW

File: src/java/com/zimbra/cs/taglib/bean/ZMessageComposeBean.java [hunk 1]
```diff
@@ -526,6 +529,43 @@ public void setSendReminderYIM(boolean mSendReminderYIM) {
...
+        return new com.zimbra.common.mailbox.Color((byte)color).toString();
+    }
+
+    public String getColorName() {
+        try {
+            int color = 0;
+            color = Integer.parseInt(mColor);
+            Color colorObj = Color.fromInt(color);
+            return colorObj.getName();
+        } catch (NumberFormatException e) {
```
Comment (fatkudu) [inline src/java/com/zimbra/cs/taglib/bean/ZMessageComposeBean.java]: can use more compact `catch (NumberFormatException | ServiceException e)`

Comment (k-kato) [reply src/java/com/zimbra/cs/taglib/bean/ZMessageComposeBean.java]: fixed it. Thank you for your comment.

File: src/java/com/zimbra/cs/taglib/bean/ZComposeUploaderBean.java [hunk 2]
```diff
@@ -337,6 +338,10 @@ private ZMessageComposeBean getComposeBean(PageContext pageContext, List<FileIte
         compose.setMessageId(getParam(F_messageid));
         compose.setDraftId(getParam(F_draftid));
 
+        int color = Byte.toUnsignedInt(com.zimbra.common.mailbox.Color.getMappedColor(getParam(F_color)));
+        compose.setColor(String.valueOf(color));
+        // compose.setRgb(null); // custom color is not supported
```
Comment (chinmay15) [inline src/java/com/zimbra/cs/taglib/bean/ZComposeUploaderBean.java]: remove commented code

Comment (k-kato) [reply src/java/com/zimbra/cs/taglib/bean/ZComposeUploaderBean.java]: removed it.

**Comment by k-kato** [comment]
rebased

**Comment by k-kato** [comment]
merged

**Pull request merged**